include/seastar/util/backtrace.hh

```diff
@@ -28,7 +28,9 @@
 #include <seastar/util/modules.hh>
 
 #ifndef SEASTAR_MODULE
+#ifdef Seastar_HAVE_EXECINFO
```

Comment:
This is so much cmake boilerplate code just to check if <execinfo.h> exists in nothing more. Since C++17, it's no longer needed - "__has_include" exists. Can we just use that and avoid all cmake changes?
since you need to know also below if execinfo exists, you can just do something like
```
#if __has_include(<execinfo.h>)
#include <execinfo.h>
#define HAVE_EXECINFO
#endif
```
and below check HAVE_EXECINFO.